aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -63,11 +65,41 @@ void gpu_index_kernel(TensorIteratorBase& iter, const IntArrayRef index_size, co
 
   if (!iter.can_use_32bit_indexing()) {
     for (auto& sub_iter : iter.with_32bit_indexing()) {
-      gpu_index_kernel(sub_iter, index_size, index_stride, f);
+      gpu_index_kernel(sub_iter, index_size, index_stride, f, is_gather_like);
     }
     return;
   }
 
+
+  char* const out_ptr = static_cast<char*>(iter.data_ptr(0));
+  char* const in_ptr = static_cast<char*>(iter.data_ptr(1));
+
+  if (is_gather_like && num_indices==1) {
+      constexpr int64_t alignment = 16;
+      //TensorIterator strides and sizes are ordered fastest moving to slowest moving,
+      //in consrast to regular sizes
+      using at::native::memory::get_alignment;
+      const size_t element_size = iter.element_size(0);
+      const auto index_element_size = iter.element_size(2);
+      // we need contiguous source and dst slices and aligned pointers and strides and slice size to do vectorized loads
+      // also we need idx to be expanded in the last dimension so we can copy entire slices
+      if (iter.ndim() == 2 && iter.strides(2)[0]==0 && iter.strides(2)[1]==index_element_size && static_cast<size_t>(iter.strides(0)[0])==element_size &&
+          static_cast<size_t>(iter.strides(1)[0])==element_size && get_alignment(out_ptr) == 16 && get_alignment(in_ptr) == 16 &&
```

Comment:
does `get_alignment(...) == 16` have anything to do with the `constexpr` above?

Reply:
good point, reworked a bit